src/modules/rust.rs

```diff
@@ -154,18 +157,12 @@ fn find_rust_toolchain_file(context: &Context) -> Option<String> {
         .map(|c| c.trim().to_owned())
     }
 
-    if let Ok(true) = context
-        .dir_contents()
-        .map(|dir| dir.has_file("rust-toolchain"))
-    {
-        if let Some(toolchain) = read_channel(Path::new("rust-toolchain")) {
-            return Some(toolchain);
-        }
-    }
-
```

Comment:
I just looked at the code again and saw that you removed this. Can you add this back again? Unlike the code below this will have the current directory cached in `context`.

Reply:
Got it. I'll fix that part.
But I've got some questions.
1. Should I copy this code for `rust-toolchain.toml` too? I'm gessing if this is for caching so this only needs to be run only once for `rust-toolnain`.
2. Should the loop below looking for parent directories start from a parent? If I add the code back, I think it would try to load the same file in the current directory twice.

Reply:
Please use this for the toml file too. You can also simplify it to something likes this:

```rust
if context
        .dir_contents()
        .map_or(false, |dir| dir.has_file("rust-toolchain"))
```

I think it's fine to have it start from the parent (though that might end up causing issues with case-insensitive file systems).

Reply:
Thanks for the quick response!
I'll use that code.
I found out that starting from the parent in the loop causes a error in the test. It seems that
 
```rust
read_channel(Path::new("rust-toolchain"), false) // returns None, path = "rust-toolchain"
```

and

```rust
let mut dir = &*context.current_dir;
read_channel(&dir.join("rust-toolchain"), false) // returns Some, path = "/tmp/**something**/rust-toolchain"
```

give different outputs. So I'll keep the loop as is.